storm-core/src/jvm/org/apache/storm/metric/api/IMetricsConsumer.java

```diff
@@ -54,6 +54,25 @@ public String toString() {
         }
         public String name; 
         public Object value;
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (!(o instanceof DataPoint)) return false;
+
+            DataPoint dataPoint = (DataPoint) o;
+
+            if (name != null ? !name.equals(dataPoint.name) : dataPoint.name != null) return false;
```

Comment:
it will be good to have parantheses around individual comparisons e.g. (name != null) or it can be simplified to StringUtils.equals(name, dataPoint.name)